src/pymatgen/core/trajectory.py

```diff
@@ -73,48 +73,61 @@ def __init__(
-                For example, for a trajectory with `M=2` and `N=4`, the
-                `site_properties` can be: [{"magmom":[5,5,5,5]}, {"magmom":[5,5,5,5]}].
+                sequence of length N, giving the properties of the N sites.
+                For example, for a trajectory with M=2 and N=4, the
+                site_properties can be: [{"magmom":[5,5,5,5]}, {"magmom":[5,5,5,5]}].
                 If a single dict, the site properties in the dict apply to all frames
-                in the trajectory. For example, for a trajectory with `M=2` and `N=4`,
+                in the trajectory. For example, for a trajectory with M=2 and N=4,
                 {"magmom":[2,2,2,2]} means that, through the entire trajectory,
                 the magmom are kept constant at 2 for all four atoms.
             frame_properties: Properties associated with the structure (e.g. total
-                energy). This should be a sequence of `M` dicts, with each dict
+                energy). This should be a sequence of M dicts, with each dict
                 providing the properties for a frame. For example, for a trajectory with
-                `M=2`, the `frame_properties` can be [{'energy':1.0}, {'energy':2.0}].
+                M=2, the frame_properties can be [{'energy':1.0}, {'energy':2.0}].
             constant_lattice: Whether the lattice changes during the simulation.
-                Should be used together with `lattice`. See usage there. This is only
+                Should be used together with lattice. See usage there. This is only
                 used for Structure-based trajectories.
-            time_step: Time step of MD simulation in femto-seconds. Should be `None`
+            time_step: Time step of MD simulation in femto-seconds. Should be None
                 for a trajectory representing a geometry optimization.
-            coords_are_displacement: Whether `coords` are given in displacements
-                (True) or positions (False). Note, if this is `True`, `coords`
+            coords_are_displacement: Whether coords are given in displacements
+                (True) or positions (False). Note, if this is True, coords
                 of a frame (say i) should be relative to the previous frame (i.e.
-                i-1), but not relative to the `base_position`.
+                i-1), but not relative to the base_position.
             base_positions: shape (N, 3). The starting positions of all atoms in the
                 trajectory. Used to reconstruct positions when converting from
                 displacements to positions. Only needs to be specified if
-                `coords_are_displacement=True`. Defaults to the first index of
-                `coords` when `coords_are_displacement=False`.
+                coords_are_displacement=True. Defaults to the first index of
+                coords when coords_are_displacement=False.
         """
+        coords = np.asarray(coords)
```

Comment:
for later reference, considered using `np.atleast3d` to normalize user input with insufficient dimensions but unfortunately when passed a 2d array, `atleast3d` inserts a singleton dim at the end rather than the start so probably more hassle than it's worth